### PR TITLE
Silence duplicate runs when editing code block node

### PR DESCRIPTION
**Pull request by pboyer**
### Purpose

**`CodeBlockNodeModel` triggers many runs which ultimately result in duplicated UI updates.** `CodeBlockNodeModel` destroys and recreates connections whenever its code is updated. So, for one change, a `CodeBlockNodeModel` triggers at least 2N + 1 node modification events where N is the number of connections to the code block. I introduced a mechanism to disable `NodeModification` events on the `HomeWorkspaceModel` containing the `NodeModel` to address this.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files

### Reviewers

@ikeough

### FYI

@aparajit-pratap @Benglin @ke-yu

### CODE REVIEW

File: src/DynamoCore/Nodes/CodeBlockNode.cs [hunk 1]
```diff
@@ -195,6 +195,10 @@ public void SetCodeContent(string newCode, ElementResolver workspaceElementResol
 
                 var inportConnections = new OrderedDictionary();
                 var outportConnections = new OrderedDictionary();
+
+                // disable node modification evnets while mutating the code
+                this.OnRequestSilenceModifiedEvents(true);
```
Comment (ikeough) [inline src/DynamoCore/Nodes/CodeBlockNode.cs]: Hooray!

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 2]
```diff
@@ -247,6 +251,38 @@ protected override void NodeModified(NodeModel node)
             }
         }
 
+        /// <summary>
+        /// Called when a node is added to the workspace and event handlers are to be added
+        /// </summary>
+        /// <param name="node">The node itself</param>
+        protected virtual void RegisterNode(NodeModel node)
```
Comment (ikeough) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: How is this not `override`? This method signature is the same as the one in `WorkspaceModel`.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 3]
```diff
@@ -247,6 +251,38 @@ protected override void NodeModified(NodeModel node)
...
+            base.RegisterNode(node);
+
+            node.RequestSilenceNodeModifiedEvents += NodeOnRequestSilenceNodeModifiedEvents;
+        }
+
+        /// <summary>
+        /// Called when a node is disposed and removed from the workspace
+        /// </summary>
+        /// <param name="node">The node itself</param>
+        protected virtual void DisposeNode(NodeModel node)
```
Comment (ikeough) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Same comment as above. Is this `override`?

**Comment by ikeough** [comment]
@pboyer Just two small comments, then LGTM. This is an exciting change. 

Does this run cleanly on the self-service CI? I'm wondering if this messes with any of the recorded tests in particular.

File: src/DynamoCore/Models/HomeWorkspaceModel.cs (hunk 3 above)
Comment (pboyer) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: Yes! Thank you.

**Comment by pboyer** [comment]
Currently awaiting CI results.

**Comment by jnealb** [comment]
QA is watching...very closely!

**Pull request merged**